Add minor display functionalities

josephroccateads opened the pull request:
This PR aims at adding some minor useful display functionalities:
- editable chart
- responsive chart
- access to plotly chart studio
- "orientation" parameter for the Legend

alexarchambault commented on render/jvm/src/main/scala/plotly/Plotly.scala:
```diff
@@ -47,7 +47,7 @@ object Plotly {
     b ++= ";\n\n  Plotly.plot('"
     b ++= div.replaceAll("'", "\\'")
-    b ++= "', data, layout);\n"
+    b ++= "', data, layout, {editable: true, responsive: true, showEditInChartStudio: true, plotlyServerURL: 'https://chart-studio.plotly.com'});\n"
```
I guess we should add a `plotly.Configuration` class, with these fields, then pass it around alongside `Layout`, so that users can enable / disable these.

alexarchambault commented on core/shared/src/main/scala/plotly/layout/Legend.scala:
```diff
@@ -15,7 +15,8 @@ import dataclass.data
       xanchor: Option[Anchor],
-      yanchor: Option[Anchor]
+      yanchor: Option[Anchor],
+  orientation: Option[String]
```
[`Orientation`](https://github.com/alexarchambault/plotly-scala/blob/89a2c04407972e234fd8c131e116b737ac8b88a3/core/shared/src/main/scala/plotly/element/Orientation.scala) could be used here instead of `String`.

josephroccateads commented:
Thanks for the comments Alexandre. You are right, I have made the changes you suggested and made some tests locally. Everything seems to be working well.

Notice that I have added the Config class and updated the required functions for its use in Almond but not in the Plotly.TraceOps.plot and Plotly.TraceSeqOps.plot as I didn't test it.

alexarchambault commented:
That looks good, thanks!

alexarchambault commented:
> Notice that I have added the Config class and updated the required functions for its use in Almond but not in the Plotly.TraceOps.plot and Plotly.TraceSeqOps.plot as I didn't test it.

No problem, this can always be added later on if needed.

alexarchambault commented:
I'd like to adjust a few things around how binary compatibility is handled, prior to cutting a release with that. Don't hesitate to ping me here if ever I forget to cut a release in the coming days.

josephroccateads commented:
Ok, perfect. Thanks Alexandre.
